newsroom/watch_lists/views.py

```diff
@@ -71,22 +71,36 @@ def create():
 @blueprint.route('/watch_lists/<_id>', methods=['GET', 'POST'])
 @login_required
 def edit(_id):
-    if not is_current_user_admin() and not is_current_user(_id):
-        flask.abort(401)
-
     watch_list = find_one('watch_lists', _id=ObjectId(_id))
     if not watch_list:
         return NotFound(gettext('Watch List not found'))
 
     if flask.request.method == 'POST':
         form = WatchListsForm(watch_list=watch_list)
         if form.validate_on_submit():
-
             updates = form.data
+            request_updates = flask.request.get_json()
+
+            # If the updates have anything other than 'users', only admin or watch_list_admin can update
+            if len(request_updates.keys()) == 1 or 'users' not in request_updates:
```

Comment:
This seem a bit generic, i.e. len(keys) == 1 or 'users'. Would it be better to check equality with `keys == ['users']`?